### ISSUE TITLE
CKAN .desktop file not properly escaping path

### ISSUE THREAD

**Issue by phardy** [KSP-CKAN/CKAN#1510]
I run KSP from Steam on a Debian Testing machine. The Desktop Entry in my `~/.local/share/applications/ckan-handler.desktop` contains

```
Exec = mono /home/peter/.local/share/Steam/steamapps/common/Kerbal Space Program/ckan.exe gui %u
```

In XFCE, if I go to the Applications menu -> Accessories and click the CKAN Launcher menu item, the following is logged to my .xsession-errors

```
Cannot open assembly '/home/peter/.local/share/Steam/steamapps/common/Kerbal': No such file or directory.
```

I guess the path needs to be quoted.

**Comment by politas** [comment]
Doesn't need to be quoted. Stripping the spaces out from around the "=" works fine, as is covered by #1498

**Comment by phardy** [comment]
It does need to be quoted, as per the freedesktop spec spaces are special characters. - http://standards.freedesktop.org/desktop-entry-spec/latest/ar01s06.html

I can handle stripping spaces from around = to fix bugs in the parser. But ignoring the quotes because of other bugs in the parser seems like a Bad Idea.

**Comment by politas** [comment]
"The Exec key must contain a command line. A command line consists of an executable program optionally followed by one or more arguments. The executable program can either be specified with its full path or with the name of the executable only. If no full path is provided the executable is looked up in the $PATH environment variable used by the desktop environment. The name or path of the executable program may not contain the equal sign ("="). Arguments are separated by a space."

See that last part? Arguments are separated by a space. The space between "mono" and "ckan.exe" fits the specification, and does not need to be quoted.

**Comment by Olympic1** [comment]
Fixed in #1536
